### PR TITLE
Added advanced Geo update menu  x-ui.sh

### PR DESCRIPTION
**Pull request by Shellgate**
A request menu is created with a submenu

### CODE REVIEW

**Comment by Shellgate** [comment]
IMG_۲۰۲۳۱۲۰۴_۰۹۳۸۵۰
IMG_۲۰۲۳۱۲۰۴_۰۹۳۹۲۸


Everything is ok 👌 but i dont add iran.dat 
Will add it to menu if you want to delete it letter you can edit it letter

**Comment by Shellgate** [comment]
IMG_۲۰۲۳۱۲۰۴_۱۰۱۰۳۵
IMG_۲۰۲۳۱۲۰۴_۱۰۱۰۵۱


The menu was fully tested and everything is correct 

Also, if you intend to remove it in the future iran.dat You can edit it, the panel was updated with both options and it works without any problems

**Comment by Enkidu-6** [comment]
A couple of things:

On lines 633 and 653 you're deleting all the .dat files ( which is not a good idea ) and then you're using wget -N, which means you're asking wget to download the files if they're newer than the local files. Well, there are no local files. You just deleted them.

Also when I choose update the geofiles, I expect them to be updated. I don't expect the files modified or anything added that wasn't in the original program. You're adding two geofiles ( named geoip_IR.dat and geosite_IR.dat ) that don't exist in the original panel and from a source that doesn't exist in the original panel. Not to mention that the Panel and users of it have no rules that uses them. So why are we adding new files to the panel and calling it an update?

If @alireza0 decides that those files are needed, that's his prerogative and he will add them to the original source but until then, in my view, we shouldn't modify them.

Last but not least, I'm not sure why you insist on adding files from jsdelivr but your menu is simply offering people a choice to update from two different sources. This implies that they are simply mirrors of the same repository and does not disclose that they will be downloading two completely different files and any rules they may already have that uses those files, may actually break.

Cheers.

**Comment by Shellgate** [comment]
There is no problem. In the first answer, I can say that instead of manually importing files, I want to delete all these files or extensions .dat are deleted 

A lot of effort has been made for these items. Do not add sieves inside these

**Comment by Enkidu-6** [comment]
> A lot of effort has been made for these items. Do not add sieves inside these

Your efforts are very much appreciated. I'm just saying that those efforts are made in the wrong direction. If I for example choose option 2 in the menu, I expect to get the newest version of Alireza's panel. I don't want to end up with Sanaei's or Marzban Panel. By the same token, option 18 should give me the newest versions of the existing Geofiles, Not a bunch of other files.

Cheers.

**Comment by Shellgate** [comment]
You should research more about these things. I have been making a new version for two days and will publish it soon

However, the current version that has been updated provides you with the most serious roles First, I clearly mentioned that all Fa.dat They will be deleted, so there is no need to name them all because they are the same b In the second case, the rolls take the latest version, the rolls of Iran are replaced instead of the Iran file in all of these, and they are complementary to the Iran file.

File: x-ui.sh [hunk 1]
```diff
@@ -612,6 +612,62 @@ ssl_cert_issue_CF() {
...
+    read -p "Please enter x-ui bin folder path. Leave blank for default. (Default: '${defaultBinFolder}')" binFolder
+    binFolder=${binFolder:-${defaultBinFolder}}
+    if [[ ! -d ${binFolder} ]]; then
+        LOGE "Folder ${binFolder} not exists!"
+        LOGI "making bin folder: ${binFolder}..."
+        mkdir -p ${binFolder}
+    fi
+    
+    cd ${binFolder}
+    rm -f *.dat
```
Comment (alireza0) [inline x-ui.sh:633]: Please don't remove the files when you are using -N swith

File: x-ui.sh [hunk 2]
```diff
@@ -612,6 +612,62 @@ ssl_cert_issue_CF() {
...
```
Comment (alireza0) [inline x-ui.sh:654]: Please don't remove the files when you are using -N swith

File: x-ui.sh [hunk 3]
```diff
@@ -612,6 +612,62 @@ ssl_cert_issue_CF() {
...
+    wget -N https://github.com/Loyalsoldier/v2ray-rules-dat/releases/latest/download/geoip.dat
+    wget -N https://github.com/Loyalsoldier/v2ray-rules-dat/releases/latest/download/geosite.dat
+    wget -O geoip_IR.dat -N https://github.com/chocolate4u/Iran-v2ray-rules/releases/latest/download/geoip.dat
+    wget -O geosite_IR.dat -N https://github.com/chocolate4u/Iran-v2ray-rules/releases/latest/download/geosite.dat
+    wget -N https://github.com/bootmortis/iran-hosted-domains/releases/latest/download/iran.dat
+    systemctl restart x-ui
```
Comment (alireza0) [inline x-ui.sh:639]: It would be nice if you can ask for restart in advance

File: x-ui.sh [hunk 4]
```diff
@@ -612,6 +612,62 @@ ssl_cert_issue_CF() {
...
+    fi
+    
+    cd ${binFolder}
+    rm -f *.dat
+    wget -N https://cdn.jsdelivr.net/gh/Loyalsoldier/v2ray-rules-dat@release/geoip.dat
+    wget -N https://cdn.jsdelivr.net/gh/Loyalsoldier/v2ray-rules-dat@release/geosite.dat
+    wget -O geoip_IR.dat -N https://cdn.jsdelivr.net/gh/chocolate4u/Iran-v2ray-rules@release/geoip.dat
+    wget -O geosite_IR.dat -N https://cdn.jsdelivr.net/gh/chocolate4u/Iran-v2ray-rules@release/geosite.dat
+    wget -N https://github.com/bootmortis/iran-hosted-domains/releases/latest/download/iran.dat 
+    systemctl restart x-ui
```
Comment (alireza0) [inline x-ui.sh:660]: It would be nice if you can ask for restart in advance.

**Review by alireza0 — CHANGES_REQUESTED** [review]
Thanks a lot. It is a kind of great job.
Please make a decision delete or using -N, and also ask for restart in advance.

**Comment by Shellgate** [comment]
I will upload the new version soon, I really worked very well and it is a perfect example

**Pull request closed without merging**